OpenAI_API/IAPIAuthentication.cs

```diff
@@ -0,0 +1,26 @@
+using System.Threading.Tasks;
+
+namespace OpenAI_API
+{
+    /// <summary>
+    /// Represents authentication to the OpenAPI API endpoint
+    /// </summary>
+    public interface IAPIAuthentication
```

Comment:
In my opinion POCO's (Plain old C# objects) shouldn't have an Interface. However this isn't a poco as it's having a method within it to validate the key.

Personally i'd extract the Validate logic to a service which can validate stuff (for now the api-key).
If you want this `APIAuthentication` to be available through DI, one should look into using the Options Pattern as this is meant to fix that problem.